src/Illuminate/Database/Query/Builder.php

```diff
@@ -2406,6 +2406,28 @@ public function doesntExist()
         return ! $this->exists();
     }
 
+    /**
+     * Execute the given callback if no rows exist for the current query.
+     *
+     * @param  \Closure $callback
+     * @return mixed
+     */
+    public function existsOr(Closure $callback)
```

Comment:
Allow all types here (should only be closure and bool, but there is some time till union types will be available).